test/built-ins/TypedArray/prototype/filter/speciesctor-destination-resizable.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2023 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.filter
+description: >
+  Throws a TypeError if new typedArray's length < count
+info: |
+  23.2.3.10 %TypedArray%.prototype.filter ( start, end )
+
+  ...
+  9. Let A be ? TypedArraySpeciesCreate(O, « count »).
+  ...
+
+  23.2.4.1 TypedArraySpeciesCreate ( exemplar, argumentList )
+
+  ...
+  3. Let result be ? TypedArrayCreate(constructor, argumentList).
+
+  23.2.4.2 TypedArrayCreate ( constructor, argumentList )
+
+  ...
+  3. If argumentList is a List of a single Number, then
+    a. If the value of newTypedArray's [[ArrayLength]] internal slot <
+    argumentList[0], throw a TypeError exception.
+  ...
+includes: [testTypedArray.js]
+features: [Symbol.species, TypedArray, resizable-arraybuffer]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  const rab1 = new ArrayBuffer(4, {"maxByteLength": 100});
+  const ta = new TA(rab1);
+  const rab2 = new ArrayBuffer(10, {"maxByteLength": 20});
+  const lengthTracking = new TA(rab2);
+  rab2.resize(0);
+  ta.constructor = { [Symbol.species]: function() { return lengthTracking; } };
+  assert.throws(TypeError, function() {
+    ta.filter(() => { return true; });
+  }
+);
+},
+[Uint8Array, Uint16Array, Uint32Array, Int8Array, Int16Array, Int32Array, Float32Array, Uint8ClampedArray] 
```

Comment:
I'd suggest that this test should be valid for all the array types defined in `typedArrayConstructors`, unless we have a good reason not to. I didn't see an obvious one, but I might've missed it.

I wonder if we should also include `BigInt64Array` and `BigUint64Array` which are not included by default in `testWithTypedArrayConstructors()`.

(Same comment for tests below)

Reply:
I realized something I didn't notice before: the BigInt typed array types are tested in separate `BigInt/` subfolders of the per-method folders, and use `testWithBigIntTypedArrayConstructors()`. So while I think we should use the default set of all non-BigInt typed arrays here, we should add copies of these tests to the `BigInt/` subfolders and edit them accordingly to use the BigInt typed array helper.